src/parquet/column/scanner-test.cc

```diff
@@ -203,42 +207,65 @@ void TestFlatScanner<FLBAType>::InitValues() {
       values_.data());
 }
 
-template<>
-void TestFlatScanner<FLBAType>::InitDescriptors(
-    std::shared_ptr<ColumnDescriptor>& d1, std::shared_ptr<ColumnDescriptor>& d2,
-    std::shared_ptr<ColumnDescriptor>& d3) {
-  NodePtr type = schema::PrimitiveNode::MakeFLBA("c1", Repetition::REQUIRED,
-      FLBA_LENGTH, LogicalType::UTF8);
-  d1.reset(new ColumnDescriptor(type, 0, 0));
-  type = schema::PrimitiveNode::MakeFLBA("c2", Repetition::OPTIONAL,
-      FLBA_LENGTH, LogicalType::UTF8);
-  d2.reset(new ColumnDescriptor(type, 4, 0));
-  type = schema::PrimitiveNode::MakeFLBA("c3", Repetition::REPEATED,
-      FLBA_LENGTH, LogicalType::UTF8);
-  d3.reset(new ColumnDescriptor(type, 4, 2));
-}
-
 typedef TestFlatScanner<FLBAType> TestFlatFLBAScanner;
 
 static int num_levels_per_page = 100;
 static int num_pages = 20;
 static int batch_size = 32;
 
-TYPED_TEST_CASE(TestFlatScanner, ParquetTypes);
+typedef ::testing::Types<BooleanType, Int32Type, Int64Type, Int96Type,
+                         FloatType, DoubleType, ByteArrayType> TestTypes;
+
+typedef TestFlatScanner<FLBAType> TestFLBAFlatScanner;
+
+TYPED_TEST_CASE(TestFlatScanner, TestTypes);
 
 TYPED_TEST(TestFlatScanner, TestScanner) {
-  this->ExecuteAll(num_pages, num_levels_per_page, batch_size);
+  this->ExecuteAll(num_pages, num_levels_per_page, batch_size, 0);
+}
+
+TEST_F(TestFLBAFlatScanner, TestScanner) {
+  this->ExecuteAll(num_pages, num_levels_per_page, batch_size, FLBA_LENGTH);
 }
 
 //PARQUET 502
 TEST_F(TestFlatFLBAScanner, TestSmallBatch) {
-  NodePtr type = schema::PrimitiveNode::MakeFLBA("c1", Repetition::REQUIRED,
-      FLBA_LENGTH, LogicalType::UTF8);
+  NodePtr type = schema::PrimitiveNode::Make("c1", Repetition::REQUIRED,
+      Type::FIXED_LEN_BYTE_ARRAY, LogicalType::DECIMAL, FLBA_LENGTH, 10, 2);
   const ColumnDescriptor d(type, 0, 0);
   MakePages(&d, 1, 100);
   InitScanner(&d);
   CheckResults(1, &d);
 }
 
+TEST_F(TestFlatFLBAScanner, TestScannerCoverage) {
```

Comment:
Can you add a comment explaining what this test case verifies? If it tests multiple things, can you break it up into multiple test cases with clear descriptions?